src/mesh/high_order_grid.cpp

```diff
@@ -1050,6 +1050,24 @@ void HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::execute_coarsen
     //}
 }
 
+template <int dim, typename real, typename MeshType, typename VectorType, typename DoFHandlerType>
+dealii::Point<dim> HighOrderGrid<dim,real,MeshType,VectorType,DoFHandlerType>::smallest_cell_coordinates()
+{
+    dealii::Point<dim> smallest_cell_coord;
+    double min_diameter = triangulation->begin_active()->diameter();
```

Comment:
The cell->diameter() returns the Triangulation straight-sided cell's diameter. When we do shape optimization, we end up moving the high-order nodes, while the Triangulation's cell vertices stay at the same place! In fact, the Triangulation cells simply becomes a connectivity map rather than anything useful physically.

Can you clarify what definition of "diameter" that you need for hexahedra?

Reply:
Nevermind, I now understand that it's just meant for the test.

Reply:
Worth keeping in mind though that it's not the actual cell sizes in more complex cases.

Reply:
Thanks, will keep it in mind (and I have introduced mapping to get cell vertex).